evaluation/red_teaming.py

```diff
@@ -133,19 +157,23 @@ def map_score_to_readable_data(results: list):
         "gandalf": False,
     }
 
-    labels = []
-    values = []
-
-    for res in results:
-        if res.score_category not in DISPLAY_LABEL_MAP or res.score_category not in EXPECTED_VALUE:
-            continue
-        labels.append(DISPLAY_LABEL_MAP[res.score_category])
-        values.append(1 if bool(res.score_value) == EXPECTED_VALUE[res.score_category] else 0)
+    values_for_all_models = {}
+    for model_name, model_result in results.items():
+        values = []
+        labels = []
+        for res in model_result:
+            if res.score_category not in DISPLAY_LABEL_MAP or res.score_category not in EXPECTED_VALUE:
+                continue
+            labels.append(DISPLAY_LABEL_MAP[res.score_category])
+            values.append(
+                1 if bool(distutils.util.strtobool(res.score_value)) == EXPECTED_VALUE[res.score_category] else 0
```

Comment:
> For starters, doesn't bool(res.score_value) do exactly what you want?

Man, Python (< 3.12) cannot natively parse a string to a boolean properly as you imagined. If I just write bool('True'), that will be a False. Using this function to parse boolean values is a very reasonable approach before Python 3.12. See reference [here](https://stackoverflow.com/a/18472142/8107122).

Reply:
> Python (< 3.12) cannot natively parse a string to a boolean properly

Thanks @GrayNekoBean, this was the answer I have been looking for which is why I insisted when Rita just adopted my stupid suggestion. Since the `score_value` is `"True"` or `"False"`, the evaluation to bool can be done "natively" with the built-in `eval`, or in a safer way with the standard lib-provided `ast.literal_eval` (the flexibility of `strtobool` is not even needed here).

Now, my point is why are we using a util from a package that's meant for something completely different to do something this simple, even if that package is part of the standard library. In the link you graciously shared, it's written in bold that distutils has been deprecated since Python 3.10 (and [actually removed in Python 3.12](https://docs.python.org/3.12/whatsnew/3.12.html#distutils)). In fact, the only reason that our tests don't break for Python 3.12 is that we happen to be installing `setuptools` (see [here](https://github.com/ucl-contoso-chat/ucl-openai-search/actions/runs/10534227591/job/29191564370?pr=26#step:6:957)), **a third-party library** that (among loads of other things) provides the now deprecated `distutils` for projects that need to maintain backward compatibility.

Is it very reasonable to introduce a deprecated dependency when you can replace all of the above with this?
```suggestion
                1 if res.score_value == str(EXPECTED_VALUE[res.score_category]) else 0
```

In fact, the only reason you created `EXPECTED_VALUE` in the first place was to use it in this specific place. With that in mind, why weren't those keys stored directly in the format you needed for your comparison (aka their actually expected values), i.e. **strings** (`"True"`, `"False"`)? Then the code could be simplified even further to sth like:

```python
1 if res.score_value == EXPECTED_VALUE[res.score_category] else 0
```

Or even better, since the function is called `map_score_to_readable_data`, why not just return `"Pass"`/`"Fail"` directly?